woocommerce/google-listings-and-ads
# Add redirects between dashboard and start

### Changes proposed in this Pull Request:

Closes #1668, #1922

If a merchant attempts to access the Dashboard before they have completed setup then they'll get a `Sorry, you are not allowed to access this page.` error message. 

![Screenshot 2023-03-20 at 18 55 52](https://user-images.githubusercontent.com/40762232/226438935-60c904e5-dce3-4b20-8928-db9617fbd879.png)

This PR adds a redirect on the backend so that if a merchant **has not** completed setup they'll be redirected to the GetStarted page when attempting to access the Dashboard.

If the merchant **has** completed setup and tries to access the GetStarted page they'll be redirected to the Dashboard.


### Detailed test instructions:

1. Checkout branch on a test site that has not yet completed GLA setup
2. Activate Google Listings & Ads on it's own and confirm you're redirected to onboarding
3. Visit `admin.php?page=wc-admin&path=/google/dashboard`
4. Confirm that you are redirected to the GetStarted page
5. Complete setup
6. Visit `admin.php?page=wc-admin&path=/google/start`
3. Confirm that you are redirected to the Dashboard

### Changelog entry

> Tweak - Redirect users between Dashboard and Get Started pages as required
